core/scripts/cli/sponsor/viewLiquidationDetails.js

```diff
@@ -1,24 +1,151 @@
+  const liquidationTimeReadable = new Date(Number(liquidation.liquidationTime.toString()) * 1000);
+  const display = `Liquidated at time ${liquidationTimeReadable} by ${liquidation.liquidator}`;
   const backChoice = "Back";
-  const withdrawAction = "Withdraw";
+  // Sponsors can withdraw rewards only if a liquidation has been disputed successfully.
+  const withdrawAction = "Dispute succeeded; withdraw rewards";
   choices = [{ name: backChoice }];
+
   // Check if the sponsor can withdraw by seeing if `withdrawLiquidation` reverts.
   try {
+    // This will only succeed if the state is PENDING_DISPUTE and a price has resolved that will make the dispute succeed.
     await emp.withdrawLiquidation.call(id, sponsorAddress);
     choices.push({ name: withdrawAction });
   } catch (err) {
-    // Withdraw wouldn't work so it shouldn't be a valid option.
+    // Withdraw wouldn't work so it shouldn't be a valid option. Let's print out a detailed response to the user to indicate why
+    // `withdraw` will fail.
+
+    // The sponsor can only withdraw rewards from a successfully disputed liquidation.
+    if (liquidation.state === LiquidationStatesEnum.PRE_DISPUTE) {
+      // If liquidation state is PRE_DISPUTE and `withdrawLiquidation` fails, then the liquidation has either expired
+      // or is pre-expiry. Only a liquidator can withdraw from an expired liquidation.
+
+      const currentTime = web3.utils.toBN(await emp.getCurrentTime());
+      const liquidationExpirationTime = web3.utils
+        .toBN(liquidation.liquidationTime)
+        .add(web3.utils.toBN(await emp.liquidationLiveness()));
+      if (liquidationExpirationTime.lte(currentTime)) {
+        console.log("Cannot withdraw rewards from a liquidation that has expired");
+      } else {
+        console.log("Cannot withdraw rewards from a liquidation that is pre-expiry and has not been disputed");
+      }
+    } else if (liquidation.state === LiquidationStatesEnum.PENDING_DISPUTE) {
+      // If the liquidation state is PENDING_DISPUTE and `withdrawLiquidation` fails, then it indicates that either
+      // a price has not resolved yet, or a price has resolved that indicates that the dispute will FAIL.
+
+      // Fetch oracle price and use it to customize error message.
+      const Finder = artifacts.require("Finder");
+      let finder = await Finder.deployed();
+      const OracleInterface = artifacts.require("OracleInterface");
+      let oracle = await OracleInterface.at(
+        await finder.getImplementationAddress(web3.utils.utf8ToHex(interfaceName.Oracle))
+      );
+      // Need to pass {from:emp.address} in this EMP view-only calls because of this web3 view method bug
+      // https://github.com/ethereum/solidity/issues/4840#issue-352539391 that would return true regardless
+      // if the Oracle has actually resolved a price. This is because `hasPrice` and `getPrice` have the
+      // `onlyRegisteredContract` modifier.
```

Comment:
nit: sorry this is actually a little more convoluted than that. You need to pass `{ from: emp.address }` because the *method will revert*.

The bug only affects what happens if you *don't* do that. One would expect web3 to throw an error, but instead it just returns true. This probably doesn't need to be mentioned in comments explicitly. I only mentioned it earlier as a warning in case you were to test it out.

Sorry if I miscommunicated that.